pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,223 @@
+						log.Errorf("localProfileTimerTask getLocalProfileConfig: %s", err)
+					}
+				}
+				ctx.ps.CheckMaxTimeTopic(wdName, "getLocalProfileConfig", start,
+					warningTime, errorTime)
+				publishZedAgentStatus(getconfigCtx)
+			}
+
+		case <-stillRunning.C:
+		}
+		ctx.ps.StillRunning(wdName, warningTime, errorTime)
+	}
+}
+
+func validateAndSetLocalProfile(localProfileBytes []byte, getconfigCtx *getconfigContext) error {
+	var localProfile = &profile.LocalProfile{}
+	err := proto.Unmarshal(localProfileBytes, localProfile)
+	if err != nil {
+		return fmt.Errorf("validateAndSetLocalProfile: Unmarshalling failed: %v", err)
+	}
+	if localProfile.GetServerToken() != getconfigCtx.profileServerToken {
+		// send something to ledmanager ??
+		return fmt.Errorf("validateAndSetLocalProfile: missamtch ServerToken for loacl profile server")
+	}
+	getconfigCtx.currentProfile = localProfile.GetLocalProfile()
+	writeReceivedProtoMessageLocalProfile(localProfileBytes)
+	publishZedAgentStatus(getconfigCtx)
+	filterAndPublishAppInstancesWithCurrentProfile(getconfigCtx)
+	return nil
+}
+
+// read saved local profile in case of particular reboot reason
+// if it is success will not read again on the next call
+func readSavedLocalProfile(getconfigCtx *getconfigContext) error {
+	if !getconfigCtx.readSavedLocalProfile && !getconfigCtx.localProfileReceived {
+		localProfile, err := readSavedProtoMessage(
+			getconfigCtx.zedagentCtx.globalConfig.GlobalValueInt(types.StaleConfigTime),
+			filepath.Join(checkpointDirname, savedLocalProfileFile), false)
+		if err != nil {
+			return fmt.Errorf("lastlocalprofile: %v", err)
+		}
+		if localProfile != nil {
+			log.Function("Using saved local profile")
+			getconfigCtx.readSavedLocalProfile = true
+			return validateAndSetLocalProfile(localProfile, getconfigCtx)
+		}
+	}
+	return nil
+}
+
+// getLocalProfileConfig connects to local profile server to fetch current profile
+func getLocalProfileConfig(url string, iteration int, getconfigCtx *getconfigContext) error {
+
+	log.Tracef("getLocalProfileConfig(%s, %d)", url, iteration)
+
+	resp, contents, err := zedcloud.SendLocal(zedcloudCtx, url, 0, nil)
+	if err != nil {
+		if err := readSavedLocalProfile(getconfigCtx); err != nil {
+			return err
+		}
+		return fmt.Errorf("SendLocal: %s", err)
+	}
+	if resp.StatusCode == http.StatusOK {
+		if err := validateProtoMessage(url, resp); err != nil {
+			// send something to ledmanager ???
+			return fmt.Errorf("getLocalProfileConfig: resp header error: %s", err)
+		}
+		getconfigCtx.localProfileReceived = true
+		return validateAndSetLocalProfile(contents, getconfigCtx)
+	}
+	return fmt.Errorf("getLocalProfileConfig: wrong response status code: %d", resp.StatusCode)
+}
+
+// filterAndPublishAppInstancesWithCurrentProfile check all app instances with currentProfile and set activate state
+func filterAndPublishAppInstancesWithCurrentProfile(getconfigCtx *getconfigContext) {
```

Comment:
That gives much better visibility to the application of the profiles.
And it might have sense to have zedmanager publish the effectiveActivate in AppInstanceStatus.

Reply:
Moved to zedmanager